docs/features/kubernetes/permission-integration.md

```diff
@@ -0,0 +1,262 @@
+            name: 'clusterOne',
+        },
+        podMetrics: [],
+        resources: [
+            {
+            type: 'pods',
+            resources: [
+                {
+                metadata: {
+                    name: 'pod1',
+                },
+                },
+            ],
+            },
+            { type: 'services', resources: [] },
+            { type: 'configmaps', resources: [] },
+            { type: 'limitranges', resources: [] },
+            {
+            type: 'deployments',
+            resources: [
+                {
+                name: 'deployment1',
+                },
+            ],
+            },
+            { type: 'replicasets', resources: [] },
+            { type: 'horizontalpodautoscalers', resources: [] },
+            { type: 'jobs', resources: [] },
+            { type: 'cronjobs', resources: [] },
+            { type: 'ingresses', resources: [] },
+            { type: 'statefulsets', resources: [] },
+            { type: 'daemonsets', resources: [] },
+        ],
+        errors: [],
+        },
+    ],
+}
+```
+
+\*Note that here we are querying a cluster that has two resouces, a pod and a deployment running
+
+### api/kubernetes/clusters
+
+```
+{
+    "items": [
+        {
+            "name": "kind",
+            "authProvider": "serviceAccount"
+        }
+    ]
+}
+```
+
+\*Note that in our app-config we use the service account method to authenticate using a service account token and we are using a local kind cluster for this example
+
+## How do the endpoints work after the permission framework integration?
+
+If we leverage the permission framework in the Kubernetes plugin this allows for finer-grained control over access to kubernetes resources via definitive and conditional policy decisions returned by a permission policy. We are allowing plugin authors the ability to restrict kubernetes endpoints so that it is safer to configure kubernetes clusters.
+
+A sample policy like
+
+```
+class DenyAllKubernetesClusterRead implements PermissionPolicy {
+  async handle(request: PolicyQuery): Promise<PolicyDecision> {
+    if (isPermission(request.permission, kubernetesClusterReadPermission)) {
+      return { result: AuthorizeResult.DENY };
+    }
+    return { result: AuthorizeResult.ALLOW };
+  }
+}
+```
+
+would now leverage the permission framework to give us the following response:
+api/kubernetes/clusters
+
+```
+{
+    "error": {
+        "name": "NotAllowedError"
+    }
+}
+```
+
+A Sample policy like
+
+```
+class K8sPermissionOnlyPodsPolicy implements PermissionPolicy {
+  async handle(
+    request: PolicyQuery,
+    user?: BackstageIdentityResponse,
+  ): Promise<PolicyDecision> {
+    if (
+      isPermission(
+        request.permission,
+        kubernetesWorkloadResourcesReadPermission,
+      )
+    ) {
+      return createKubernetesConditionalDecision(
+        request.permission,
+        kubernetesConditions.isOfKind({ kind: 'pods' } ?? { kind: '' }),
```

Comment:
What happens if you change this sample policy to return a compound PermissionCriteria, like this?
```suggestion
        {
          anyOf: [
            kubernetesConditions.isOfKind({ kind: 'pods' }),
            kubernetesConditions.isOfKind({ kind: 'deployments' }),
          ],
        }
```

Reply:
Unit test was added for complex permission criteria and logic was added to endpoint to handle the Allof, anyof, and not members of the permission criteria.